### PR TITLE
[DOCS] Document `include_in_*` nested mapping parms

### LINKED ISSUE #52309: [DOCS] Mapping options `include_in_root` and `include_in_parent` are not documented
The options `include_in_root` and `include_in_parent` allow you to modify how a nested object is indexed in the root document or a parent document respectively:

https://github.com/elastic/elasticsearch/blob/1dc9dd42235e876fde8c3b31934f1bc88d668896/server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java#L226-L235

I cannot find any mention of them in today's documentation, although they were [documented in 1.7](https://www.elastic.co/guide/en/elasticsearch/reference/1.7/mapping-nested-type.html).

### PR DESCRIPTION
**Pull request by jrodewig**
Adds documentation for the `include_in_parent` and `include_in_root`
mapping parameters for the `nested` mapping datatype.

Closes #52309.

### CODE REVIEW

File: docs/reference/mapping/types/nested.asciidoc [hunk 1]
```diff
@@ -183,17 +183,26 @@ or <<request-body-search-stored-fields, `stored_fields`>>.
...
-    <<mapping-types,datatype>>, including `nested`. New properties
-    may be added to an existing nested object.
+(Optional, object)
+The fields within the nested object, which can be of any
+<<mapping-types,datatype>>, including `nested`. New properties
+may be added to an existing nested object.
+
+`include_in_parent`::
+(Optional, boolean)
+If `true`, inner objects are indexed as <<object,flat key-value pairs>> for the
```
Comment (jtibshirani) [inline docs/reference/mapping/types/nested.asciidoc]: To me, this description could be hard to understand. An alternate suggestion: "If `true`, all fields in the nested object are also added to the parent document as standard (flat) fields." A similar suggestion applies to `include_in_root` below.

**Review by jtibshirani — COMMENTED** [review]
Thanks for addressing this!

File: docs/reference/mapping/types/nested.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/mapping/types/nested.asciidoc]: Thank you for the feedback @jtibshirani. I've reworded as suggested with 817c0ffcfca.

File: docs/reference/mapping/types/nested.asciidoc [hunk 2]
```diff
@@ -183,18 +183,26 @@ or <<request-body-search-stored-fields, `stored_fields`>>.
...
+may be added to an existing nested object.
+
+`include_in_parent`::
+(Optional, boolean)
+If `true`, all fields in the nested object are also added to the parent document
+as standard (flat) fields. Defaults to `false`.
+
+`include_in_root`::
+(Optional, boolean)
+If `true`, all fields in the nested object are also added to the topmost
```
Comment (jtibshirani) [inline docs/reference/mapping/types/nested.asciidoc]: Small comment, maybe 'root document' would be a little clearer.

**Review by jtibshirani — APPROVED** [review]
Looks good to me.

**Comment by jrodewig** [comment]
Thanks @jtibshirani.

**Comment by jrodewig** [comment]
### Backport commits
master 12ed6f1
7.x 1a14ae4e1b5
7.6 7c887035bdbeecd5293abe19ce5e77f2e90620e6

**Pull request merged**